core/src/smartcontracts/isi/expression.rs

```diff
@@ -328,9 +405,25 @@ impl Evaluate for RaiseTo {
         wsv: &WorldStateView,
         context: &Context,
     ) -> Result<Self::Value, Self::Error> {
-        let left = self.left.evaluate(wsv, context)?;
-        let right = self.right.evaluate(wsv, context)?;
-        Ok(left.pow(right).into())
+        use NumericValue::*;
+        let value = self.left.evaluate(wsv, context)?;
+        let exp = self.right.evaluate(wsv, context)?;
+        match (value, exp) {
+            (U32(value), U32(exp)) => value
+                .checked_pow(exp)
+                .ok_or(Error::Math(MathError::Overflow))
+                .map(NumericValue::from),
+            (U128(value), U32(exp)) => value
```

Comment:
there is u128 and u32, is it ok?
```suggestion
            (U128(value), U128(exp)) => value
```

Reply:
Yea, this is intentional you can look at [u128::checked_pow](https://doc.rust-lang.org/std/primitive.u128.html#method.checked_pow). `u32` is more then enough to cover all possible exponent  values that do not cause overflow.